File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.ts
function searchRange(nums: number[], target: number): number[] {
    function binarySearch(nums: number[], target: number, is_left: boolean= true): number{
        let left:number = 0, right:number = nums.length-1, index:number =-1;

        while(left<=right){
            let mid:number= Math.floor((left+right)/2);
            if(nums[mid]> target){
                right = mid-1;
            }
            else if(nums[mid]< target){
                left = mid+1;
            }
            else{
                index = mid;
                if(is_left){
                    right= mid-1;
                }
                else{
                    left= mid+1;
                }
            }
        }
        return index;    
    }
    let left = binarySearch(nums, target, true);
    let right = binarySearch(nums, target, false);
    return [left, right];
    
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution(object):
    def searchRange(self, nums, target):
        def binary_search(nums, target, is_right):
            left = 0;
            right = len(nums)-1
            index = -1;

            while left<=right:
                mid = (left+right)/2

                if nums[mid]< target:
                    left = mid +1
                elif nums[mid]> target:
                    right = mid-1;
                else:
                    index = mid
                    if is_right:
                        left = mid+1
                    else:
                        right = mid-1

            return index
        right = binary_search(nums, target, True)
        left = binary_search(nums, target, False)

        return[left, right]
        